media/js/src/GraphMapping.js

```diff
@@ -122,6 +122,103 @@ const exportGraph = function(state) {
     return obj;
 };
 
+const convertGraph = function(json) {
+    return {
+        gId: json.id,
+        gTitle: json.title || '',
+        gSummary: json.summary,
+        gInstructions: json.instructions,
+        gInstructorNotes: json.instructor_notes,
+        gType: json.graph_type,
+        gAssignmentType: json.assignment_type,
+        gIsPublished: json.is_published,
+        gIsFeatured: json.featured,
+        gDisplayFeedback: json.display_feedback,
+        gDisplayShadow: json.display_shadow,
+        gNeedsSubmit: json.needs_submit,
+        gTopic: json.topic,
+
+        gShowIntersection: json.show_intersection,
+        gDisplayIntersection1: json.display_intersection_1,
+        gDisplayIntersection2: json.display_intersection_2,
+        gDisplayIntersection3: json.display_intersection_3,
+        gIntersectionLabel: json.intersection_label,
+        gIntersection2Label: json.intersection_2_label,
+        gIntersection3Label: json.intersection_3_label,
+
+        gIntersectionHorizLineLabel: json.intersection_horiz_line_label,
+        gIntersectionVertLineLabel: json.intersection_vert_line_label,
+        gIntersection2HorizLineLabel: json.intersection_2_horiz_line_label,
+        gIntersection2VertLineLabel: json.intersection_2_vert_line_label,
+        gIntersection3HorizLineLabel: json.intersection_3_horiz_line_label,
+        gIntersection3VertLineLabel: json.intersection_3_vert_line_label,
+
+        gLine1Slope: window.parseFloat(json.line_1_slope) || 0,
```

Comment:
For number parsing like this, econplayground handles this using our internal [forceFloat](https://github.com/ccnmtl/econplayground/blob/master/media/js/src/utils.js#L194) function, so we can use that instead here.

Also, using functions off the `window` namespace is generally discouraged in a react application like this, unless absolutely necessary - it just makes the code less flexible, i.e. `window` may not be available in node.js testing environments:

```
~ $ node
Welcome to Node.js v18.1.0.
Type ".help" for more information.
> parseFloat
[Function: parseFloat]
> window.parseFloat
Uncaught ReferenceError: window is not defined
> 
```

Reply:
I converted all the number strings to integer/float values, but I needed to modify the check at line 303 in the Question Editor as a result: `forceFloat` wouldn't return the string values if the evaluated property was a label, name, or boolean.